### ISSUE TITLE
F-Strings with private variables

### ISSUE THREAD

**Issue by AsylumET** [renpy/renpy#5442]
I have encountered some peculiar behavior while using f-strings. In the first code snip we have a very simple ren'py game script. It should print 3 lines in the console. Instead it errors with the f-string in read1. I have tested to ensure the private variables are accessible and clearly, they are. I also ran only the python code as a standalone python script and all 3 work as expected. See the 2nd code snip for that.

script.rpy
```
# The game starts here.
init python:
    class OddFStringExample:
        def __init__(self):
            self.__FPATH = "filepath/"
            self.__FEXT = ".txt"
        def read1(self, name):
            the_file = f"{self.__FPATH}{name}{self.__FEXT}"
            print(the_file)
        def read2(self, name):
            the_file = self.__FPATH+name+self.__FEXT
            print(the_file)
        def read3(self, name):
            thep = self.__FPATH
            thex = self.__FEXT
            the_file = f"{thep}{name}{thex}"
            print(the_file)

label start:
    python:
        ## This errors
#        readit = OddFStringExample()
#        readit.read1("test_file1")
        ## This works
        readit = OddFStringExample()
        readit.read2("test_file2")
        ## And this works as well
        readit = OddFStringExample()
        readit.read3("test_file3")
    "And we are done"
    # This ends the game.
    return

# Ren'Py console shows
#filepath/test_file2.txt
#filepath/test_file3.txt
```

myscript.py
```
## Running as a standalone python script with the latest stable version
## the output is as expected:
#filepath/test_file1.txt
#filepath/test_file2.txt
#filepath/test_file3.txt
class OddFStringExample:
    def __init__(self):
        self.__FPATH = "filepath/"
        self.__FEXT = ".txt"
    def read1(self, name):
        the_file = f"{self.__FPATH}{name}{self.__FEXT}"
        print(the_file)
    def read2(self, name):
        the_file = self.__FPATH+name+self.__FEXT
        print(the_file)
    def read3(self, name):
        thep = self.__FPATH
        thex = self.__FEXT
        the_file = f"{thep}{name}{thex}"
        print(the_file)

## This errors
readit = OddFStringExample()
readit.read1("test_file1")
## This works
readit = OddFStringExample()
readit.read2("test_file2")
## And this works as well
readit = OddFStringExample()
readit.read3("test_file3")
```

and finally, the error produced:

```
I'm sorry, but an uncaught exception occurred.

While running game code:
  File "game/script.rpy", line 20, in script
    python:
  File "game/script.rpy", line 23, in <module>
    readit.read1("test_file1")
  File "game/script.rpy", line 8, in read1
    the_file = f"{self.__FPATH}{name}{self.__FEXT}"
AttributeError: 'OddFStringExample' object has no attribute '_OddFStringExample__FPATH'

-- Full Traceback ------------------------------------------------------------

Full traceback:
  File "game/script.rpy", line 20, in script
    python:
  File "C:\PythonStuff\renpy-8.2.0-sdk\renpy\ast.py", line 823, in execute
    renpy.python.py_exec_bytecode(self.code.bytecode, self.hide, store=self.store)
  File "C:\PythonStuff\renpy-8.2.0-sdk\renpy\python.py", line 1178, in py_exec_bytecode
    exec(bytecode, globals, locals)
  File "game/script.rpy", line 23, in <module>
    readit.read1("test_file1")
  File "game/script.rpy", line 8, in read1
    the_file = f"{self.__FPATH}{name}{self.__FEXT}"
AttributeError: 'OddFStringExample' object has no attribute '_OddFStringExample__FPATH'

Windows-10-10.0.22621 AMD64
Ren'Py 8.2.0.24012702
Sand 1.0
Sat Mar 30 02:51:54 2024
```

**Comment by Gouvernathor** [comment]
I know what's happening.
First, renpy reads the file and replaces variables starting with \_\_ with versions incorporating the file name and not starting with two underscores anymore, *except* for the insides of strings (including f-strings) which is privileged.
Then Python parsing comes along, and inside class bodies, converts the two-underscore names it finds into ones incorporating the class name.
Then, the ast substitution recently-ish added to fix the case of f-strings replaces the remaining cases, which apparently means f-strings outside class bodies.

The issue is that the python munging apparently happens before the ast magic, something we didn't foresee...
We could un-munge f-string fields inside class bodies, using double ast magic, before re-munging them using the existing f-string ast magic ?
The only other solution I can see is upgrading the file reader to make it understand f-string fields, something I already tried and failed at the time.

**Comment by renpytom** [comment]
So, the issue here is that Ren'Py isn't applying it's __name handling inside string. That's fixed - it now applies it everywhere, including inside strings. 

The fix - which will come out in 8.3 - has a side effect, as it means that "__foo" will become "_m1_script__foo" everywhere in script.rpy. So: 

   print(f"__foo = {__foo}}")

will become:

    print(f"_m1_script__foo = {_m1_script__foo}")

I think this is fine. Strings like __word just don't happen very often in text. What worries me slightly is this happening in filenames - but even there, double-underscore-word should be quite rare, so I'm willing to accept the slight breakage.
